testing/jormungandr-integration-tests/src/networking/p2p.rs

```diff
@@ -334,63 +343,156 @@ pub fn node_put_itself_in_preffered_layers() {
+    const INTERVAL_SECS: u64 = 3;
 
-    let low_topic_of_interests = TopicsOfInterest {
-        messages: "low".to_owned(),
-        blocks: "low".to_owned(),
-    };
-
-    let mut network_controller = network::builder()
-        .star_topology(SERVER, vec![fast_client_alias, slow_client_alias])
+    let mut network_controller = NetworkBuilder::default()
+        .single_trust_direction(CLIENT, SERVER)
         .initials(vec![
-            wallet("delegated0").with(1_000_000).delegated_to(SERVER),
-            wallet("delegated1")
+            WalletTemplateBuilder::new("delegated1")
                 .with(1_000_000)
-                .delegated_to(fast_client_alias),
-            wallet("delegated2")
+                .delegated_to(CLIENT),
+            WalletTemplateBuilder::new("delegated2")
                 .with(1_000_000)
-                .delegated_to(slow_client_alias),
-        ])
-        .custom_config(vec![
-            SpawnParams::new(fast_client_alias).topics_of_interest(high_topic_of_interests),
-            SpawnParams::new(slow_client_alias).topics_of_interest(low_topic_of_interests),
+                .delegated_to(SERVER),
         ])
         .build()
         .unwrap();
 
-    let _server = network_controller.spawn_and_wait(SERVER);
-    let fast_client = network_controller.spawn_and_wait(fast_client_alias);
-    let slow_client = network_controller.spawn_and_wait(fast_client_alias);
+    let _server = network_controller
+        .spawn_custom(
+            network_controller
+                .spawn_params(SERVER)
+                .gossip_interval(Duration::new(INTERVAL_SECS, 0)),
+        )
+        .unwrap();
 
-    process_utils::sleep(30);
+    let client = network_controller
+        .spawn_custom(
+            network_controller
+                .spawn_params(CLIENT)
+                .log_level(Level::TRACE),
+        )
+        .unwrap();
 
-    let fast_client_block_recv_cnt = fast_client
-        .rest()
-        .stats()
-        .unwrap()
-        .stats
-        .unwrap()
-        .block_recv_cnt;
-    let slow_client_block_recv_cnt = slow_client
-        .rest()
-        .stats()
-        .unwrap()
-        .stats
-        .unwrap()
-        .block_recv_cnt;
+    process_utils::sleep(10);
+
+    let log_timestamps: Vec<u64> = client
+        .log_content()
+        .into_iter()
+        .filter(|s| s.contains("accept_gossips"))
+        .skip(1)
+        .map(|t| parse_timestamp(&t))
+        .collect();
+
+    let mut prev = None;
+
+    for log_timestamp in log_timestamps {
+        match prev {
+            None => prev = Some(log_timestamp),
+            Some(prev) => {
+                assert!(log_timestamp - prev >= INTERVAL_SECS);
+            }
+        }
+    }
+}
 
-    assert!(
-        fast_client_block_recv_cnt > slow_client_block_recv_cnt,
-        "node with high block topic of interest should have more recieved blocks"
+#[ignore = "long runner"]
+#[test]
+fn network_stuck_check() {
```

Comment:
for long running tests there is a feature: `soak-non-functional`.  
```
#[cfg(feature = "soak-non-functional")]
```

Reply:
Nice, thanks!